### PR TITLE
fix(core): fix wrong calculation of the color of the sky

### LINKED ISSUE #652: Sky turns black when low altitude and high tilt
Hello,

Trying iTowns, with only the ortho. As I tilt the map, the sky suddently renders black instead of being blue:
image

Reproduced in Chrome with the following code:
```
var positionOnGlobe = { longitude: 2.471311, latitude: 49.258868, altitude: 25 };
// iTowns namespace defined here
var viewerDiv = document.getElementById('viewerDiv');
var globeView = new itowns.GlobeView(viewerDiv, positionOnGlobe);
globeView.addLayer(ORTHO);
globeView.addEventListener(itowns.GLOBE_VIEW_EVENTS.GLOBE_INITIALIZED, function(){
	console.log("itowns.GLOBE_VIEW_EVENTS.GLOBE_INITIALIZED");
globeView.controls.setTilt(80, true);
});
```

As you zoom out, the sky turns back to blue

Fabien

### PR DESCRIPTION
**Pull request by gchoqueux**
simplification of the calculation formula by a linear calculation

fix #652

### CODE REVIEW

File: src/Core/Prefab/GlobeView.js [hunk 1]
```diff
@@ -247,6 +247,12 @@ function GlobeView(viewerDiv, coordCarto, options = {}) {
 
     const renderer = this.mainLoop.gfxEngine.renderer;
 
+    const coordCam = new Coordinates(this.referenceCrs, 0, 0, 0);
```
Comment (zarov) [inline src/Core/Prefab/GlobeView.js:250]: No need to set the CRS to `referenceCRS` as it is getting rewritten to `EPSG:4978` after.

File: src/Core/Prefab/GlobeView.js [hunk 2]
```diff
@@ -256,19 +262,24 @@ function GlobeView(viewerDiv, coordCarto, options = {}) {
...
-            renderer.setClearColor(0x030508, renderer.getClearAlpha());
+        // get altitude camera
+        coordCam.set('EPSG:4978', this.camera.camera3D.position).as('EPSG:4326', coordGeoCam);
+        const altitude = coordGeoCam.altitude();
+
+        // if altitude camera is below a threshold (limitAlti)
+        // to fade the color of space to the color of the sky proportional to the normalized interval [ground, threshold]
+        if (altitude < limitAlti) {
+            const t = (limitAlti - altitude) / limitAlti;
+            const colorSky = spaceColor.clone().lerp(skyBaseColor, t);
```
Comment (zarov) [inline src/Core/Prefab/GlobeView.js]: Do we really need to clone each time ? What about adding a global `const skyColor` and `copy()` inside it each time ?

**Review by zarov — COMMENTED** [review]
Seems good to me, and way simpler ! Thank you for this fix.

@fnicollet, could you test this branch to confirm it is okay with you ?

File: src/Core/Prefab/GlobeView.js [hunk 3]
```diff
@@ -256,19 +262,24 @@ function GlobeView(viewerDiv, coordCarto, options = {}) {
...
-            var color = new THREE.Color(0x93d5f8);
-            renderer.setClearColor(color.multiplyScalar(1.0 - t), renderer.getClearAlpha());
-        } else if (len >= lim) {
-            renderer.setClearColor(0x030508, renderer.getClearAlpha());
+        // get altitude camera
+        coordCam.set('EPSG:4978', this.camera.camera3D.position).as('EPSG:4326', coordGeoCam);
+        const altitude = coordGeoCam.altitude();
+
+        // if altitude camera is below a threshold (limitAlti)
+        // to fade the color of space to the color of the sky proportional to the normalized interval [ground, threshold]
```
Comment (autra) [inline src/Core/Prefab/GlobeView.js]: > If the camera altitude is below limitAlti, we interpolate between the sky color and the space color

File: src/Core/Prefab/GlobeView.js (hunk 1 above)
Comment (gchoqueux) [reply src/Core/Prefab/GlobeView.js:250]: crs is mandatory parameter in constructor

**Comment by gchoqueux** [comment]
@zarov @autra branch updated

File: src/Core/Prefab/GlobeView.js (hunk 2 above)
Comment (gchoqueux) [reply src/Core/Prefab/GlobeView.js]: yes

**Review by zarov — APPROVED** [review]
Good to me, thanks.

File: src/Core/Prefab/GlobeView.js [hunk 4]
```diff
@@ -256,19 +262,24 @@ function GlobeView(viewerDiv, coordCarto, options = {}) {
...
         this.fogDistance = mfogDistance * Math.pow((len - size * 0.99) * 0.25 / size, 1.5);
 
-        if (len < lim) {
-            var t = Math.pow(Math.cos((lim - len) / (lim - v.x * size * 0.9981) * Math.PI * 0.5), 1.5);
-            var color = new THREE.Color(0x93d5f8);
-            renderer.setClearColor(color.multiplyScalar(1.0 - t), renderer.getClearAlpha());
-        } else if (len >= lim) {
-            renderer.setClearColor(0x030508, renderer.getClearAlpha());
+        // get altitude camera
+        coordCam.set('EPSG:4978', this.camera.camera3D.position).as('EPSG:4326', coordGeoCam);
```
Comment (autra) [inline src/Core/Prefab/GlobeView.js]: `EPSG:4978` should be referenceCRS

File: src/Core/Prefab/GlobeView.js (hunk 3 above)
Comment (autra) [reply src/Core/Prefab/GlobeView.js]: What about this comments? At least let's correct the English :-)

**Review by autra — CHANGES_REQUESTED** [review]
still some little nits, but not much :-)

File: src/Core/Prefab/GlobeView.js [hunk 5]
```diff
@@ -256,19 +263,24 @@ function GlobeView(viewerDiv, coordCarto, options = {}) {
...
+        const altitude = coordGeoCam.altitude();
+
+        // If the camera altitude is below limitAlti,
+        // we interpolate between the sky color and the space color
```
Comment (autra) [inline src/Core/Prefab/GlobeView.js]: Oh missed it, please ignore my other comment.

File: src/Core/Prefab/GlobeView.js (hunk 4 above)
Comment (gchoqueux) [reply src/Core/Prefab/GlobeView.js]: yes, PR updated

**Review by autra — APPROVED** [review]

**Comment by autra** [comment]
Thanks!

**Pull request merged**